src/api_types.jl

```diff
@@ -104,222 +104,316 @@ abstract  type Hmpih end
+    H5P.FILE_CREATE      = read_const(:H5P_CLS_FILE_CREATE_ID_g)
+    H5P.FILE_ACCESS      = read_const(:H5P_CLS_FILE_ACCESS_ID_g)
+    H5P.DATASET_CREATE   = read_const(:H5P_CLS_DATASET_CREATE_ID_g)
+    H5P.DATASET_ACCESS   = read_const(:H5P_CLS_DATASET_ACCESS_ID_g)
+    H5P.DATASET_XFER     = read_const(:H5P_CLS_DATASET_XFER_ID_g)
+    H5P.FILE_MOUNT       = read_const(:H5P_CLS_FILE_MOUNT_ID_g)
+    H5P.GROUP_CREATE     = read_const(:H5P_CLS_GROUP_CREATE_ID_g)
+    H5P.GROUP_ACCESS     = read_const(:H5P_CLS_GROUP_ACCESS_ID_g)
+    H5P.DATATYPE_CREATE  = read_const(:H5P_CLS_DATATYPE_CREATE_ID_g)
+    H5P.DATATYPE_ACCESS  = read_const(:H5P_CLS_DATATYPE_ACCESS_ID_g)
+    H5P.STRING_CREATE    = read_const(:H5P_CLS_STRING_CREATE_ID_g)
+    H5P.ATTRIBUTE_CREATE = read_const(:H5P_CLS_ATTRIBUTE_CREATE_ID_g)
+    H5P.OBJECT_COPY      = read_const(:H5P_CLS_OBJECT_COPY_ID_g)
+    H5P.LINK_CREATE      = read_const(:H5P_CLS_LINK_CREATE_ID_g)
+    H5P.LINK_ACCESS      = read_const(:H5P_CLS_LINK_ACCESS_ID_g)
+
+    H5T.STD_I8LE        = read_const(:H5T_STD_I8LE_g)
+    H5T.STD_I8BE        = read_const(:H5T_STD_I8BE_g)
+    H5T.STD_U8LE        = read_const(:H5T_STD_U8LE_g)
+    H5T.STD_U8BE        = read_const(:H5T_STD_U8BE_g)
+    H5T.STD_I16LE       = read_const(:H5T_STD_I16LE_g)
+    H5T.STD_I16BE       = read_const(:H5T_STD_I16BE_g)
+    H5T.STD_U16LE       = read_const(:H5T_STD_U16LE_g)
+    H5T.STD_U16BE       = read_const(:H5T_STD_U16BE_g)
+    H5T.STD_I32LE       = read_const(:H5T_STD_I32LE_g)
+    H5T.STD_I32BE       = read_const(:H5T_STD_I32BE_g)
+    H5T.STD_U32LE       = read_const(:H5T_STD_U32LE_g)
+    H5T.STD_U32BE       = read_const(:H5T_STD_U32BE_g)
+    H5T.STD_I64LE       = read_const(:H5T_STD_I64LE_g)
+    H5T.STD_I64BE       = read_const(:H5T_STD_I64BE_g)
+    H5T.STD_U64LE       = read_const(:H5T_STD_U64LE_g)
+    H5T.STD_U64BE       = read_const(:H5T_STD_U64BE_g)
+    H5T.IEEE_F32LE      = read_const(:H5T_IEEE_F32LE_g)
+    H5T.IEEE_F32BE      = read_const(:H5T_IEEE_F32BE_g)
+    H5T.IEEE_F64LE      = read_const(:H5T_IEEE_F64LE_g)
+    H5T.IEEE_F64BE      = read_const(:H5T_IEEE_F64BE_g)
+    H5T.C_S1            = read_const(:H5T_C_S1_g)
+    H5T.STD_REF_OBJ     = read_const(:H5T_STD_REF_OBJ_g)
+    H5T.STD_REF_DSETREG = read_const(:H5T_STD_REF_DSETREG_g)
+    H5T.NATIVE_B8       = read_const(:H5T_NATIVE_B8_g)
+    H5T.NATIVE_INT8     = read_const(:H5T_NATIVE_INT8_g)
+    H5T.NATIVE_UINT8    = read_const(:H5T_NATIVE_UINT8_g)
+    H5T.NATIVE_INT16    = read_const(:H5T_NATIVE_INT16_g)
+    H5T.NATIVE_UINT16   = read_const(:H5T_NATIVE_UINT16_g)
+    H5T.NATIVE_INT32    = read_const(:H5T_NATIVE_INT32_g)
+    H5T.NATIVE_UINT32   = read_const(:H5T_NATIVE_UINT32_g)
+    H5T.NATIVE_INT64    = read_const(:H5T_NATIVE_INT64_g)
+    H5T.NATIVE_UINT64   = read_const(:H5T_NATIVE_UINT64_g)
+    H5T.NATIVE_FLOAT    = read_const(:H5T_NATIVE_FLOAT_g)
+    H5T.NATIVE_DOUBLE   = read_const(:H5T_NATIVE_DOUBLE_g)
+
+    nothing
+end
+__init_globals__()
```

Comment:
```suggestion
 ```
Typo here?

Reply:
Not a typo exactly — just left-over cruft. At some point while rewriting, it was necessary explicitly initialize everything while the module was being defined, but that's no longer the case.